cleanco/clean.py

```diff
@@ -111,10 +115,23 @@ def custom_basename(name, terms, suffix=True, prefix=False, middle=False, **kwar
                     del nnparts[idx + 1]
                     del nparts[idx + 1]
 
+    for country_name in country_names:
+        if nparts and nparts[-1].lower() == country_name.lower():
+            # In specific cases (e.g. "XXX of Sweden"), we should leave the country name
+            if len(nparts) >= 2 and nparts[-2] == 'of':
+                continue
+
+            nnparts = nnparts[:-1]
+            nparts = nparts[:-1]
+
     return strip_tail(" ".join(nparts))
 
 
-def basename(name, suffix=True, prefix=True, middle=False):
+# convenience for most common use cases that don't parametrize base name extraction
+def basename(name, suffix=True, prefix=True, middle=False, country=None):
```

Comment:
I would also name this something like `country_code` to avoid confusion with the name.

Reply:
This works with both country codes and country names.
`country_codes.get(country, country)` will transform the code into a country name, or leave it as a country name.

Reply:
Oh, smart! 🧠 Maybe a small comment, as that is kind of hidden if you read over it?